sdk/src/quic.rs

```diff
@@ -6,3 +6,8 @@ pub const QUIC_MAX_UNSTAKED_CONCURRENT_STREAMS: usize = 128;
 
 pub const QUIC_MAX_TIMEOUT_MS: u32 = 2_000;
 pub const QUIC_KEEP_ALIVE_MS: u64 = 1_000;
+
+// Based on commonly-used handshake timeouts for various TCP
+// applications. Different applications vary, but most seem to
+// be in the 30-60 second range
+pub const QUIC_CONNECTION_HANDSHAKE_TIMEOUT_MS: u64 = 60_000;
```

Comment:
i don't think we have 60s to sit around twirling our fingers.  do the cited applications have a human on one end of the connection rather than two machines?  something like 3s max seems more reasonable

Reply:
TCP has high timeouts mainly for historical reasons and the wide variety of networks TCP runs on.

What do we do if the connection attempt failed? Try again?

Reply:
The quic client in this case just returns an error when the handshake times out, but at least some places that use the quic client, such as forwarding, simply ignore the error. They do keep track of the number of packets successfully sent, but if there's a failure, there's no attempt to resend as far as I can tell.

Reply:
So your advice on the best course of action here would be greatly appreciated.

Reply:
ignoring errors is a bug

Reply:
Given that this is an existing bug, and that it will take additional non-trivial work to implement resend (possibly to a different peer) for forwarding (and any other places that need it), I think we should leave this for another PR. @leoluk for now, what is your recommended handshake timeout?

Reply:
There's no generic recommendation - depends on whether we want to fail early or keep retrying. For "fail early", ~3s seems reasonable.

Reply:
In that case, since we currently don't have any infrastructure to deal with an early fail, I'm leaning toward keeping a longer, more conservative timeout, to allow Quic to implement a rudimentary "retry" of a sort, and also due to the fact that right now we're not completely sure about the characteristics of the network - we have nodes all over the world. That being said, I definitely agree with the sentiment that we should eventually move toward handling send errors and handling resend explicitly on our end - and, as we learn more about the characteristics of the network, shorten the handshake timeout.

Reply:
was it confirmed that quinn actually retries while awaiting a `Connecting`?

Reply:
After some digging, it appears the client and server just author the handshake packets, and then their transmit/retransmit are handled just like any other packet.